src/libraries/System.Text.Encoding.CodePages/src/System/Text/EncodingCharBuffer.cs

```diff
@@ -72,7 +72,10 @@ internal unsafe bool AddChar(char ch)
         internal unsafe bool AddChar(char ch1, char ch2, int numBytes)
         {
             // Need room for 2 chars
-            if (_charEnd - _chars < 2)
+            // It is possible for `_chars` to be null, and in such cases, `_charEnd` will also be null.
```

Comment:
The overflow trick is hard to reason about. Would it be better to explicitly check for `_chars != null` like we do it in the other (more common) `AddChar` overload above?

Reply:
Done!